homeassistant/components/mqtt/climate.py

```diff
@@ -604,22 +603,8 @@ async def _set_climate_attribute(
         await self._publish(cmnd_topic, payload)
         return changed
 
-    @abstractmethod
-    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
-        """Set hvac mode."""
-
-    @abstractmethod
-    async def async_set_operation_mode(self, operation_mode: str) -> None:
-        """Set operation mode."""
-
-    async def async_set_temperature(self, **kwargs: Any) -> None:
-        """Set new target temperatures."""
-        operation_mode: HVACMode | str | None
-        if (operation_mode := kwargs.get(ATTR_HVAC_MODE)) is not None:
-            await self.async_set_hvac_mode(operation_mode)
-        if (operation_mode := kwargs.get(ATTR_OPERATION_MODE)) is not None:
-            await self.async_set_operation_mode(operation_mode)
```

Comment:
I hoped we could leave this one in the shared part

Reply:
I can put that back.
If the `async_set_hvac_mode` and `async_set_operation_mode` are defined as abstract, would it be best to implement both on the climate and water heater and raise a `NotImplementedError` for the unused one?

Reply:
Yeah, raise `NotImplementedError` and make them abstract.